cerberus/kubernetes/client.py

```diff
@@ -0,0 +1,86 @@
+from kubernetes import client, config
+from kubernetes.client.rest import ApiException
+import logging
+
+
+# Load kubeconfig and initialize kubernetes python client
+def initialize_clients(kubeconfig_path):
+    global cli
+    config.load_kube_config(kubeconfig_path)
+    cli = client.CoreV1Api()
+
+
+# List nodes in the cluster
+def list_nodes():
+    nodes = []
+    try:
+        ret = cli.list_node(pretty=True)
+    except ApiException as e:
+        logging.error("Exception when calling CoreV1Api->list_node: %s\n" % e)
+    for node in ret.items:
+        nodes.append(node.metadata.name)
+    return nodes
+
+
+# List pods in the given namespace
+def list_pods(namespace):
+    pods = []
+    try:
+        ret = cli.list_namespaced_pod(namespace, pretty=True)
+    except ApiException as e:
+        logging.error("Exception when calling \
+                       CoreV1Api->list_namespaced_pod: %s\n" % e)
+    for pod in ret.items:
+        pods.append(pod.metadata.name)
+    return pods
+
+
+# Monitor the status of the cluster nodes and set the status to true or false
+def monitor_nodes():
+    nodes = list_nodes()
+    ready_nodes = []
+    notready_nodes = []
+    for node in nodes:
+        try:
+            node_info = cli.read_node_status(node, pretty=True)
+        except ApiException as e:
+            logging.error("Exception when calling \
+                           CoreV1Api->read_node_status: %s\n" % e)
+        node_status = node_info.status.conditions[-1].status
+        if node_status == "True":
+            ready_nodes.append(node)
```

Comment:
It doesn't look like we are actually using ready_nodes for anything at this point and we don't return it. Do we even need to collect it?

Reply:
Right, we are collecting it to use if we need it as part of the report or something in the future but our main focus is on the failures, so we can skip it like you mentioned @dry923. Will update it. Thanks.

Reply:
@dry923 fixed it, thanks.